docs/hlo_passes.md

```diff
@@ -0,0 +1,165 @@
+# HLO Passes 
+
+[XLA (Accelerated Linear Algebra)](https://openxla.org/xla) is an open source compiler for machine learning that is part of the [OpenXLA](http://openxla.org) ecosystem. 
+
+The XLA compiler takes models from popular frameworks such as PyTorch, TensorFlow, and JAX, and optimizes the models for high-performance execution across different hardware platforms including GPUs, CPUs, and Google’s custom accelerator, the TPU. Among other uses, XLA serves as a high-performance backend to Tensorflow and is the only method of running code on the TPU. 
+
+Programs running on XLA are described in a high-level language called HLO (High Level Optimizer) which also serves as the machine-independent intermediate representation (IR) in the compiler. XLA performs numerous critical optimizations at the HLO-level before lowering the program into a machine-instruction level IR for code generation. This document outlines an overview of the HLO parts of the XLA compiler including summaries of important optimization and analysis passes.
+
+[StableHLO](https://openxla.org/stablehlo), which is also part of the OpenXLA ecosystem, is not covered in the scope of this document. For information on StableHLO passes see [OpenXLA \- StableHLO \- StableHLO Passes](https://openxla.org/stablehlo/generated/stablehlo_passes).
+
+## Features and Overview 
+
+In HLO, programs are represented as dataflow graphs where values in the graph are arrays similar to Tensorflow’s program representation. Individual HLO instructions describe high-level operations performed on arrays such as convolution or element-wise operations such as add and subtract. A comprehensive description of HLO Operations can be found at [OpenXLA \- XLA \- Operation Semantics](https://openxla.org/xla/operation_semantics), which represents isolated functional computations given to the compiler.
+
+XLA includes a large number of optimizations and transformations which operate on HLO. At the time of writing there are several hundred HLO optimization and analysis passes identified. These passes take advantage of the high abstraction level of HLO to perform optimizations and other transformations which are more difficult to perform on a program representation closer to the abstraction level of machine instructions.
+
+An HLO operation generally lowers to a loop nest which computes the array output of the operation. However, HLO does not represent these loop nests explicitly. Instead, only the shapes of arrays are explicitly specified. These shapes determine the bounds of each loop and the depth of the loop nest after lowering. Focusing only on the shape and semantics of the operation makes many operations easier. 
```

Comment:
Lots of this information seems redundant with:

- https://openxla.org/xla/terminology
- https://openxla.org/xla/architecture

I'd recommend removing this from the "passes" page since this info is more about terminology / architecture. If you see gaps in those documents perhaps we can update it there